File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> ans;

    int zero = 0, same= 0;
    //배열의 크기가 작으므로 O(N^2)으로 푼다.
    //크다면 정렬 후 이분탐색으로 해야할듯.

    for(auto l : lottos){
        if(l == 0) {
            zero +=1;
            continue;
        }

        for(auto n : win_nums){
            if( n == l )
                same += 1;
        }
    }

    ans.push_back(zero == 0 && same == 0 ? 6 : 7 - (zero + same));
    ans.push_back(same == 0 ? 6 : 7 - same);

    return ans;
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int zeroCnt=0, cnt=0;

        for (int num:lottos) {
            if(num == 0) zeroCnt+=1;

            for (int winNum:win_nums) {
                if(winNum == num){
                    cnt+=1;
                    break;
                }
            }
        }

        answer[0] = (zeroCnt+cnt) < 2 ? 6 : 7-(zeroCnt+cnt);
        answer[1] = (cnt) < 2 ? 6 : 7-cnt;

        return answer;
    }
}